File: src/main/java/com/diluv/api/utils/UserUtilities.java
package com.diluv.api.utils;

import com.diluv.api.DiluvAPI;
import org.jooq.Record5;
import org.jooq.Record6;

import java.sql.Timestamp;
import java.util.HashMap;
import java.util.Map;

import static com.diluv.api.models.Tables.USER;

public class UserUtilities {
    public static Map<String, Object> getUserByUserId(long userId, boolean authorized) {
        Record5<String, String, String, Timestamp, Integer> user = DiluvAPI.getDSLContext().select(USER.USERNAME, USER.EMAIL, USER.AVATAR, USER.CREATED_AT, USER.PERMISSION)
                .from(USER)
                .where(USER.ID.eq(userId))
                .fetchOne();

        Map<String, Object> userOut = new HashMap<>();
        if (user != null) {
            userOut.put("username", user.get(USER.USERNAME));
            userOut.put("avatar", user.get(USER.AVATAR));
            userOut.put("createdAt", user.get(USER.CREATED_AT));

            if (authorized) {
                userOut.put("permission", user.get(USER.PERMISSION));
                userOut.put("email", user.get(USER.EMAIL));
            }
        }
        return userOut;
    }

    public static Map<String, Object> getUserSettingsByUserId(long userId) {
        Record6<String, String, String, String, String, Boolean> user = DiluvAPI.getDSLContext().select(USER.EMAIL, USER.AVATAR, USER.FIRST_NAME, USER.LAST_NAME, USER.LOCATION, USER.MFA_ENABLED)
                .from(USER)
                .where(USER.ID.eq(userId))
                .fetchOne();

        Map<String, Object> userSettingsOut = new HashMap<>();

        if (user != null) {
            userSettingsOut.put("email", user.get(USER.EMAIL));
            userSettingsOut.put("avatar", user.get(USER.AVATAR));
            userSettingsOut.put("firstName", user.get(USER.FIRST_NAME));
            userSettingsOut.put("lastName", user.get(USER.LAST_NAME));
            userSettingsOut.put("location", user.get(USER.LOCATION));
            userSettingsOut.put("mfaEnabled", user.get(USER.MFA_ENABLED));
        }
        return userSettingsOut;
    }
}

File: src/main/kotlin/com/diluv/api/utils/UserUtilities.kt
package com.diluv.api.utils

import com.diluv.api.models.Tables.USER
import org.jooq.SQLDialect
import org.jooq.impl.DSL
import java.sql.Connection

fun Connection.getUserByUserId(userId: Long, authorized: Boolean): Map<String, Any> {
    val transaction = DSL.using(this, SQLDialect.MYSQL)

    val user = transaction.select(USER.USERNAME, USER.EMAIL, USER.AVATAR, USER.CREATED_AT, USER.PERMISSION)
            .from(USER)
            .where(USER.ID.eq(userId))
            .fetchOne()

    if (user != null) {
        var userOut = mapOf<String, Any>(
                "username" to user.get(USER.USERNAME),
                "avatar" to user.get(USER.AVATAR),
                "createdAt" to user.get(USER.CREATED_AT),
                "projects" to this.getProjectsByUserId(userId)
        )

        if (authorized) {
            userOut += mapOf<String, Any>(
                    "permission" to user.get(USER.PERMISSION),
                    "email" to user.get(USER.EMAIL)
            )
        }
        return userOut
    } else {
        return mapOf()
    }
}

fun Connection.getUserSettingsByUserId(userId: Long): Map<String, Any> {
    val transaction = DSL.using(this, SQLDialect.MYSQL)

    val user = transaction.select(USER.EMAIL, USER.AVATAR, USER.FIRST_NAME, USER.LAST_NAME, USER.LOCATION, USER.MFA_ENABLED)
            .from(USER)
            .where(USER.ID.eq(userId))
            .fetchOne()

    if (user != null) {
        return mapOf<String, Any>(
                "email" to user.get(USER.EMAIL),
                "avatar" to user.get(USER.AVATAR),
                "firstName" to user.get(USER.FIRST_NAME),
                "lastName" to user.get(USER.LAST_NAME),
                "location" to user.get(USER.LOCATION),
                "mfaEnabled" to user.get(USER.MFA_ENABLED)
        )
    }
    return mapOf()
}